src/hotspot/share/runtime/continuationFreezeThaw.cpp

```diff
@@ -630,13 +621,14 @@ void FreezeBase::freeze_fast_existing_chunk() {
 
     freeze_fast_copy(chunk, chunk_start_sp CONT_JFR_ONLY(COMMA false));
   } else { // the chunk is empty
-    DEBUG_ONLY(_empty = true;)
-    const int chunk_start_sp = chunk->sp();
+    const int chunk_start_sp = chunk->stack_size() - frame::metadata_words_at_top;
```

Comment:
Do we need the minus frame::metadata_words_at_top? Since the chunk is empty I would expect the chunk_start_sp to be the same as for the new chunk case which is just chunk->stack_size().

Reply:
I think you are right; I changed it to chunk->stack_size() as you propose, and adjusted the bottom to remove the metadata part instead. I think this looks nicer.